app/src/main/java/com/movie/app/main/MainActivity.kt

```diff
@@ -1,29 +1,30 @@
 package com.movie.app.main
 
 import android.os.Bundle
-import android.support.v7.widget.DividerItemDecoration
-import android.support.v7.widget.LinearLayoutManager
 import android.view.View
 import android.widget.AdapterView
 import android.widget.ArrayAdapter
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.chad.library.adapter.base.BaseQuickAdapter
 import com.movie.app.BaseActivity
 import com.movie.app.R
 import com.movie.app.details.DetailsMovieActivity
-import com.movie.app.modules.Movie
 import com.movie.app.modules.MovieSortType
+import com.movie.app.modules.MoviesResult
+import com.movie.app.util.Result
 import com.movie.app.util.notifyVisibleItems
 import com.movie.app.util.setDefaultColor
 import com.movie.app.util.setToolbar
 import kotlinx.android.synthetic.main.activity_main.*
 import kotlinx.android.synthetic.main.toolbar_main_activity.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
 import timber.log.Timber
-import javax.inject.Inject
 
-class MainActivity : BaseActivity(), MainActivityContractor.View {
+class MainActivity : BaseActivity() {
 
-    @Inject
-    lateinit var presenter: MainActivityContractor.Presenter
+    val viewModel: MainViewModel  by viewModel()
```

Comment:
**Suggestion:** Consider using a ViewModelFactory for injecting the ViewModel, instead of directly using the 'by viewModel()' delegation. This would make the code more testable and decoupled.
```suggestion
    private val viewModelFactory: ViewModelProvider.Factory
    val viewModel: MainViewModel by viewModels { viewModelFactory }
```